lib/compiler.js

```diff
@@ -538,3 +525,61 @@ function resolveRelativePaths(relativePaths, appRootDir) {
     }
   });
 }
+
+function isPreferredExtension (filename, disallowedExtensions) {
+  disallowedExtensions = disallowedExtensions || [];
+
+  var includeExtensions = require.extensions;
+  var excludeExtensions = {};
+  excludeExtensions[FILE_EXTENSION_JSON] = FILE_EXTENSION_JSON;
+
+  disallowedExtensions.forEach(function(ext) {
+    excludeExtensions[ext] = ext;
+  });
+
+  var ext = path.extname(filename);
+  return (ext in includeExtensions) && !(ext in excludeExtensions);
+}
+
+function fixFileExtension (filepath, files, disallowedExtensions,
+  contentDataType) {
```

Comment:
> a single parameter like fixOptions

Sounds good, I think the name `options` is good enough.

Few more thoughts: `.node` should be ignored everywhere we ignore `.json`, it's just another extensions that gets special treatment from Node.js.

If find `contentDataType: 'function'` difficult to understand in the context of the calling code. Perhaps `onlyScriptsExportingFunction: true` would be a better option name?